http-server-jetty/src/main/java/io/micronaut/servlet/jetty/JettyFactory.java

```diff
@@ -352,10 +379,11 @@ private void applyAdditionalPorts(Server server, ServerConnector serverConnector
             if (CollectionUtils.isNotEmpty(exposedPorts)) {
                 for (Integer exposedPort : exposedPorts) {
                     if (!exposedPort.equals(serverConnector.getLocalPort())) {
-                        ServerConnector connector = new ServerConnector(
-                            server,
-                            serverConnector.getConnectionFactories().toArray(ConnectionFactory[]::new)
-                        );
+                        ServerConnector connector = new ServerConnector(server,
+                                exposedPort.equals(endpointDefaultConfiguration.getPort().orElse(-1))
+                                    ? serverConnector.getConnectionFactories().stream().filter(x -> !x.getProtocol().equals("SSL")).toArray(ConnectionFactory[]::new)
+                                    : serverConnector.getConnectionFactories().toArray(ConnectionFactory[]::new)
```

Comment:
we need to add a configuration option, I don't think default to `HTTP` if it is the management port makes sense